### PR TITLE
Feature/kak/add map view#10

### LINKED ISSUE #10: Add map view
Add map activity/fragment: [wireframe](https://marvelapp.com/4c295bb/screen/37237006)

Display map, create header bar. Open map view when map button clicked from place view toolbar.

Do not implement toolbar buttons yet, except for back/up navigation.

Do net yet load destination markers.

Depends on #7.

### PR DESCRIPTION
**Pull request by flibbertigibbet**
## Overview

Add a map view with a toolbar and filter button bar, with overlay. Current user location is displayed as a blue dot.

## Demo

screenshot_1525120997

## Notes

In implementing #11, the map will need to know if it is to display events or destinations, and so should set the activity title text and up navigation target appropriately (so, go back to places list instead of home, if showing destinations.)

@lederer I'm not sure what the Material Design "view day" button to the left of the search on the [map page wireframe](https://marvelapp.com/4c295bb/screen/37237006) is going to do here. Could you remind me what we should do with that please?

## Testing

 - Follow the instructions added to the `README` for setting up a maps API key
 - Tap the map icon in the toolbar on either the places or events list
 - Map activity should open
 - Filter overlay should function as expected (open and close, filtering not implemented)
 - Current user location as used for distances should show as a blue dot on the map

Closes #10.

### CODE REVIEW

**Comment by lederer** [comment]
> I'm not sure what the Material Design "view day" button to the left of the search on the map page wireframe is going to do here.

That button toggles between list and map view. When in list view, the button is a map icon. Tapping it goes to map view. When in map view, the button is a list icon (we're coopting the "view day" icon here to mean list, bc it looks more like a list of cards than material's official list icon does). Tapping it goes to list view.

The button to the left of that one is the places/events toggle. When in places mode, the button is an event icon. When in event mode, the button is a map marker icon.

**Comment by maurizi** [comment]
Since we have a dedicated button for switching between the map & list view, presumably the up-button in the upper left should go back to the homescreen, rather than the event/place list view?

**Comment by lederer** [comment]
> Since we have a dedicated button for switching between the map & list view, presumably the up-button in the upper left should go back to the homescreen, rather than the event/place list view?

Yes. Good point. In essence, the map and list view are intended to feel like they're just two different modes of the same screen, and so the back arrow should return to the home view in either case.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,11 @@
 # cac-tripplanner-android
 GoPhillyGo Android app
+
+
+## Setup
+
+Copy API keys file for debug and release builds:
+ - `cp example/google_maps_api.xml app/src/debug/res/values/google_maps_api.xml`
+ - `cp example/google_maps_api.xml app/src/release/res/values/google_maps_api.xml`
```
[the remark this answers is not in the document]
Comment (flibbertigibbet) [reply README.md]: Yup, that's what happened, good catch.

**Comment by flibbertigibbet** [comment]
Updated; this should be ready for another look.

I discovered that the Play services library could be updated to the latest version, but that it was necessary to disable Instant Run in Android Studio and clear the cache to avoid a build error after changing the library version.

**Comment by maurizi** [comment]
Merging for @flibbertigibbet since she is away.

**Pull request merged**